# Исследование надёжности заёмщиков

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

## Шаг 1. Откроем файл с данными и изучим общую информацию

Импортируем библиотеку Pandas. Прочитаем файл data.csv и сохраним его в переменной data

In [1]:
import pandas as pd
data = pd.read_csv('/datasets/data.csv')

Получим общую информацию о данных таблицы data

In [2]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


Получим первые 15 строк таблицы

In [3]:
data.head(15)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   

     total_income                            purpose  
0   253875.639453                      покупка жилья  
1   112080.014102            приобретение автомобиля  
2   145885.952297                      покупка жилья  
3   267628.550329         дополнительное образование  
4   158616.077870                    сыграть свадьбу  
5   255763.565419                      покупка жилья  
6   240525.971920                  операции с жильем  
7   135823.934197                        образование  
8    95856.832424              на проведение свадьбы  
9   144425.938277            покупка жилья для семьи  
10  113943.491460               покупка недвижимости  
11   77069.234271  покупка коммерческой недвижимости  
12            NaN                    сыграть свадьбу  
13  130458.228857            приобретение автомобиля  
14  165127.911772         покупка жилой недвижимости

Рассмотрим уникальные значения данных для каждого столбца

In [4]:
for row in data:
    print(data[row].value_counts())

 0     14149
 1      4818
 2      2055
 3       330
 20       76
-1        47
 4        41
 5         9
Name: children, dtype: int64
-327.685916     1
-1580.622577    1
-4122.460569    1
-2828.237691    1
-2636.090517    1
               ..
-7120.517564    1
-2146.884040    1
-881.454684     1
-794.666350     1
-3382.113891    1
Name: days_employed, Length: 19351, dtype: int64
35    617
40    609
41    607
34    603
38    598
42    597
33    581
39    573
31    560
36    555
44    547
29    545
30    540
48    538
37    537
50    514
43    513
32    510
49    508
28    503
45    497
27    493
56    487
52    484
47    480
54    479
46    475
58    461
57    460
53    459
51    448
59    444
55    443
26    408
60    377
25    357
61    355
62    352
63    269
64    265
24    264
23    254
65    194
66    183
22    183
67    167
21    111
0     101
68     99
69     85
70     65
71     58
20     51
72     33
19     14
73      8
74      6
75      1
Name: dob_years, dtype: int64
среднее   

**Вывод**

Всего в таблице 12 столбцов:
1. children — количество детей в семье;
2. days_employed — общий трудовой стаж в днях;
3. dob_years — возраст клиента в годах;
4. education — уровень образования клиента;
5. education_id — идентификатор уровня образования;
6. family_status — семейное положение;
7. family_status_id — идентификатор семейного положения;
8. gender — пол клиента;
9. income_type — тип занятости;
10. debt — имел ли задолженность по возврату кредитов;
11. total_income — ежемесячный доход;
12. purpose — цель получения кредита.


Каждая строка таблицы содержит информацию о клиенте банка. Общее количество значений в наших данных 21525.
После изучения общей информации о данных выявлены следующие семь проблем, которые нужно решить:
1. В столбце "children" встречается ошибка в знаке (-1 - отрицательное количество детей), аномально большое значение для нашей выборки (20 детей);
2. В столбце "days_employed" встречается ошибка в знаке (отрицательное количество дней трудового стажа) и пропущенные значения, что говорит о том, что по некоторым клиентам доступна не вся информация;
3. В столбце "dob_years" встречаются нулевые значения;
4. В столбце "education" встречается разный регистр букв в словах;
5. В столбце "gender" встречаются строки с пропущенными данными;
6. В столбце "total_income" встречаются пропущенные значения, неудобный формат типа данных (значения типа float), который лучше преобразовать в int;
7. В столбце "purpose" много дубликатов с похожими или идентичными целями получения кредита.

## Шаг 2. Предобработка данных

### Обработка пропусков

Исправим ошибочные значения в столбце количество детей ("children")

In [5]:
data['children'] = data['children'].replace(20, 2)
data['children'] = data['children'].replace(-1, 1)

Приведем к единому формату регистра столбец уровень образования ("education") методом str.lower() и выведем уникальные значения с помощью метода unique()

In [6]:
data['education'] = data['education'].str.lower()
data['education'].unique()

array(['высшее', 'среднее', 'неоконченное высшее', 'начальное',
       'ученая степень'], dtype=object)

Узнаем количество пропущенных значений в столбце пол клиента ("gender")

In [7]:
data[data['gender'] == 'XNA'].count()

children            1
days_employed       1
dob_years           1
education           1
education_id        1
family_status       1
family_status_id    1
gender              1
income_type         1
debt                1
total_income        1
purpose             1
dtype: int64

Так как всего одна строка с пропущенным значением, удалим её

In [8]:
data = data[data['gender'] != 'XNA']

Для заполнения нулевых значений в столбце возраст клиента ("dob_years") посчитаем медиану возраста клиента для каждого типа занятости и посчитаем нулевые значения методом isnull().sum()

In [9]:
age_medians = data.groupby('income_type')['dob_years'].median()
age_medians

data.loc[(data['income_type'] == 'пенсионер') & (data['dob_years'] == 0), 'dob_years'] = 60
data.loc[(data['income_type'] == 'студент') & (data['dob_years'] == 0), 'dob_years'] = 22
data.loc[(data['income_type'] == 'сотрудник') & (data['dob_years'] == 0), 'dob_years'] = 39
data.loc[(data['income_type'] == 'предприниматель') & (data['dob_years'] == 0), 'dob_years'] = 42.5
data.loc[(data['income_type'] == 'компаньон') & (data['dob_years'] == 0), 'dob_years'] = 39
data.loc[(data['income_type'] == 'госслужащий') & (data['dob_years'] == 0), 'dob_years'] = 40
data.loc[(data['income_type'] == 'в декрете') & (data['dob_years'] == 0), 'dob_years'] = 39
data.loc[(data['income_type'] == 'безработный') & (data['dob_years'] == 0), 'dob_years'] = 38

data['dob_years'].isnull().sum()

0

Для заполнения пропусков проверим типы данных и длины столбцов. Метод info() позволяет посмотреть сводную информацию о нашей таблице

In [10]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21524 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21524 non-null  int64  
 1   days_employed     19350 non-null  float64
 2   dob_years         21524 non-null  float64
 3   education         21524 non-null  object 
 4   education_id      21524 non-null  int64  
 5   family_status     21524 non-null  object 
 6   family_status_id  21524 non-null  int64  
 7   gender            21524 non-null  object 
 8   income_type       21524 non-null  object 
 9   debt              21524 non-null  int64  
 10  total_income      19350 non-null  float64
 11  purpose           21524 non-null  object 
dtypes: float64(3), int64(4), object(5)
memory usage: 2.1+ MB
None


Так как для некоторых клиентов не указан общий трудовой стаж ("days_employed") и ежемесячный доход ("total_income"), заполним пропущенные значения на основе имеющихся данных. Для получения среднего арифметического значения вызовем метод mean()

In [11]:
income_grouped_mean = data.groupby('income_type')['total_income'].mean() 
print(income_grouped_mean)

income_type
безработный        131339.751676
в декрете           53829.130729
госслужащий        170898.309923
компаньон          202417.136353
пенсионер          137127.465690
предприниматель    499163.144947
сотрудник          161380.260488
студент             98201.625314
Name: total_income, dtype: float64


Применим median() для расчета медианы

In [12]:
income_grouped_median = data.groupby('income_type')['total_income'].median()
print(income_grouped_median)

income_type
безработный        131339.751676
в декрете           53829.130729
госслужащий        150447.935283
компаньон          172319.266339
пенсионер          118514.486412
предприниматель    499163.144947
сотрудник          142594.396847
студент             98201.625314
Name: total_income, dtype: float64


Так как данные среднего значения и медианы совпадают, то в соотвествующих столбцах ("days_employed" и "total_income") заполним пропуски на медианные значения, а в столбце общий трудовой стаж ("days_employed") преобразуем отрицательные значения в положительные используя функцию abs()

In [13]:
def fillbygroup(data, row):
    unique_inc_type = data['income_type'].unique()
    for type in unique_inc_type:
        data.loc[data['income_type'] == type, row] = data.loc[data['income_type'] == type, row].fillna(data[data['income_type'] == type]['total_income'].median())
    return data

data = fillbygroup(data, 'total_income')

data['days_employed'] = data['days_employed'].apply(abs)
data = fillbygroup(data, 'days_employed')
data.info()
data

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21524 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21524 non-null  int64  
 1   days_employed     21524 non-null  float64
 2   dob_years         21524 non-null  float64
 3   education         21524 non-null  object 
 4   education_id      21524 non-null  int64  
 5   family_status     21524 non-null  object 
 6   family_status_id  21524 non-null  int64  
 7   gender            21524 non-null  object 
 8   income_type       21524 non-null  object 
 9   debt              21524 non-null  int64  
 10  total_income      21524 non-null  float64
 11  purpose           21524 non-null  object 
dtypes: float64(3), int64(4), object(5)
memory usage: 2.1+ MB


children  days_employed  dob_years education  education_id  \
0             1    8437.673028       42.0    высшее             0   
1             1    4024.803754       36.0   среднее             1   
2             0    5623.422610       33.0   среднее             1   
3             3    4124.747207       32.0   среднее             1   
4             0  340266.072047       53.0   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1    4529.316663       43.0   среднее             1   
21521         0  343937.404131       67.0   среднее             1   
21522         1    2113.346888       38.0   среднее             1   
21523         3    3112.481705       38.0   среднее             1   
21524         2    1984.507589       40.0   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

        total_income                       purpose  
0      253875.639453                 покупка жилья  
1      112080.014102       приобретение автомобиля  
2      145885.952297                 покупка жилья  
3      267628.550329    дополнительное образование  
4      158616.077870               сыграть свадьбу  
...              ...                           ...  
21520  224791.862382             операции с жильем  
21521  155999.806512          сделка с автомобилем  
21522   89672.561153                  недвижимость  
21523  244093.050500  на покупку своего автомобиля  
21524   82047.418899         на покупку автомобиля  

[21524 rows x 12 columns]

In [14]:
data['days_employed'].abs()

0          8437.673028
1          4024.803754
2          5623.422610
3          4124.747207
4        340266.072047
             ...      
21520      4529.316663
21521    343937.404131
21522      2113.346888
21523      3112.481705
21524      1984.507589
Name: days_employed, Length: 21524, dtype: float64

**Вывод**

После изучения общей информации о данных были разрешены следующие проблемы:

1. Так как у большинства клиентов численность детей варьируется от 0 до 5, то в столбце "children" были исправлены ошибки в знаке (отрицательное количество детей), аномально большое значение для нашей выборки (20 детей);
2. В столбце "days_employed" исправлена ошибка в знаках (отрицательное количество дней трудового стажа) и заполнены пропуски на медианные значения;
3. В столбце "dob_years" нулевые значения были заполнены на основании медианы для каждого типа дохода;
4. В столбце "education" регистр букв был приведен к единому формату;
5. В столбце "gender" удалили строку с пропущенным значением;
6. В столбце "total_income" пропуски были заполнены на медианные значения.

Теперь таблица готова к дальнейшему изучению: ненужные пропуски удалены, но при этом сохранены важные данные.
Причины появления отрицательных значений и попусков в данных можно уточнить у сотрудника, который выгружал эти данные. Возможно, что это связано со сбоем в системе сбора информации.

### Замена типа данных

Получим информацию о типах данных в таблице

In [15]:
data.dtypes

children              int64
days_employed       float64
dob_years           float64
education            object
education_id          int64
family_status        object
family_status_id      int64
gender               object
income_type          object
debt                  int64
total_income        float64
purpose              object
dtype: object

В колонках общий трудовой стаж ("days_employed"), ежемесячный доход ("total_income") и возраст клиента ("dob_years") встречаются значения типа float (вещественное число), поэтому преобразуем значения столбцов в тип данных int методом astype()

In [16]:
data['days_employed'] = data['days_employed'].astype('int')
data['total_income'] = data['total_income'].astype('int')
data['dob_years'] = data['dob_years'].astype('int')
data.dtypes

children             int64
days_employed        int64
dob_years            int64
education           object
education_id         int64
family_status       object
family_status_id     int64
gender              object
income_type         object
debt                 int64
total_income         int64
purpose             object
dtype: object

**Вывод**

Здесь мы заменили вещественный тип данных на целочисленный в колонках общий трудовой стаж ("days_employed"), ежемесячный доход ("total_income") и возраст клиента ("dob_years"). В нужный тип данных (целое число) нам позволил преобразовать значения метод astype().

### Обработка дубликатов

Найдем дубликаты в столбце "purpose" (цель получения кредита) методом value_counts(). Возвращаем количество дубликатов методом duplicated().sum()). Для удаления дубликатов используем метод drop_duplicates()

In [17]:
purpose_counts = data['purpose'].value_counts() 
#print(purpose_counts)

print('Количество дубликатов в данных:', data.duplicated().sum())

data = data.drop_duplicates().reset_index(drop = True)
print('Количество дубликатов в данных:', data.duplicated().sum())

Количество дубликатов в данных: 71
Количество дубликатов в данных: 0


**Вывод**

Количество дубликатов, которое мы обнаружили: 71 шт. Причины появления дубликатов нужно выяснить, так как причиной возникновения этой ошибки может быть как человеческий фактор, так и сбой. Для удаления дубликатов применялся метод drop_duplicates(), так как его целью является поиск идентичных строк.

### Лемматизация

Импортируем библиотеку pymystem3 и collections

In [18]:
from pymystem3 import Mystem
from collections import Counter
m = Mystem()

Посчитаем все варианты целей получения кредита

In [19]:
data['purpose'].value_counts()

свадьба                                   791
на проведение свадьбы                     768
сыграть свадьбу                           765
операции с недвижимостью                  675
покупка коммерческой недвижимости         661
операции с жильем                         652
покупка жилья для сдачи                   651
операции с коммерческой недвижимостью     650
покупка жилья                             646
жилье                                     646
покупка жилья для семьи                   638
строительство собственной недвижимости    635
недвижимость                              633
операции со своей недвижимостью           627
строительство жилой недвижимости          624
покупка своего жилья                      620
покупка недвижимости                      620
строительство недвижимости                619
ремонт жилью                              607
покупка жилой недвижимости                606
на покупку своего автомобиля              505
заняться высшим образованием      

Выделим ключевые слова, которые отражают цель получения кредита: "свадьба", "недвижимость", "жилье", "автомобиль", "образование". Напишем функцию, которая проведет лемматизацию и заменим полученный список лемм на ключевые слова

In [20]:
categories = ['свадьба', 'недвижимость', 'жилье', 'автомобиль', 'образование']

def lemmatize(text):
    lemma = m.lemmatize(text)
    for word in categories:
        if word in lemma:
            lemma = word
    return lemma

data['purpose_group'] = data['purpose'].apply(lemmatize)
data['purpose_group'].value_counts()

недвижимость    6350
жилье           4460
автомобиль      4306
образование     4013
свадьба         2324
Name: purpose_group, dtype: int64

Проведем замену, так как жилье тоже можно отнести к недвижимости 

In [21]:
data.loc[data['purpose_group'] == 'жилье', 'purpose_group'] = 'недвижимость'
data['purpose_group'].value_counts()

недвижимость    10810
автомобиль       4306
образование      4013
свадьба          2324
Name: purpose_group, dtype: int64

**Вывод**

Были посчитаны все варианты целей получения кредита, после чего выделили ключевые слова (так как многие цели названы похожим образом, но при этом имеют один и тот же смысл). Таким образом, получили четыре категории, которые отражают цель получения кредита: "свадьба", "недвижимость", "автомобиль", "образование". Провели лемматизацию, заменив список лемм на ключевые слова. Те цели, которые содержали производные слова от слова "жилье" заменили на категорию недвижимость.

### Категоризация данных

Для ответов на вопросы нам необходимо было провести категоризацию данных по целям кредита, ежемесячному доходу и наличию детей. Проведем категоризацию по ежемесячному доходу, написав функцию, которая разбивает доходы клиентов по квантилям (25%, 50% и 75%)

In [22]:
statistics = data['total_income'].describe()
statistics[4:7]

25%    107620.0
50%    142594.0
75%    195818.0
Name: total_income, dtype: float64

Запишем функцию для определения дохода по категориям

In [23]:
def division_income_group(income):
    if income <= statistics[4]: return 1
    elif statistics[4] < income <= statistics[5]: return 2
    elif statistics[5] < income <= statistics[6]: return 3
    else: return 4

data['income_group'] = data['total_income'].apply(division_income_group)
data['income_group'].value_counts()

2    5479
1    5364
4    5363
3    5247
Name: income_group, dtype: int64

В результате категоризации данных по ежемесячному доходу мы получили, что:
1. Ежемесячный доход 5364 клиентов меньше или равен 107 620 руб. 
2. Ежемесячный доход 5479 клиентов в интервале от 107 620 руб. до 142 594 руб.
3. Ежемесячный доход 5247 клиентов в интервале от 142 594 руб. до 195 818 руб.
4. Ежемесячный доход 5363 клиентов больше 195 818 руб.

Запишем функцию для для категоризации клиентов по наличию детей, где: 1 - есть дети, 0 - нет детей.

In [24]:
def determine_children(children):
    if children > 0: return 1
    else: return 0

data['child_exist'] = data['children'].apply(determine_children)

**Вывод**

Для ответов на вопросы в предыдушем разделе мы провели категоризацию данных по целям кредита, а в этом - по ежемесячному доходу и количеству детей. Проводя категоризацию по ежемесячному доходу, написали функцию, которая разбивает доходы клиентов по квантилям (25%, 50% и 75%), а категоризируя данные по количеству детей вводим бинарную переменную.

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

## Шаг 3. Ответим на вопросы

In [25]:
data_pivot = data.pivot_table(index = ['child_exist'], values = 'debt').round(3)
#data_pivot['ratio'] = data_pivot[1] / data_pivot[0]
data_pivot.head()

debt
child_exist       
0            0.075
1            0.092

**Вывод**

Зависимость есть, наличие детей у клиента увеличивает вероятность просроченной задолженности (0,092 > 0,075). Возможно, это связано с тем, что в нашей выборке клиенты предпочитают тратить деньги на детей, чем возвращать их в срок.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [26]:
data_pivot = data.pivot_table(index = ['family_status'], columns = 'debt', values = 'gender', aggfunc = 'count')

data_pivot['ratio'] = round(data_pivot[1] / (data_pivot[0] + data_pivot[1]), 3)
data_pivot.sort_values('ratio', ascending = False)

debt                       0    1  ratio
family_status                           
Не женат / не замужем   2536  274  0.098
гражданский брак        3762  388  0.093
женат / замужем        11408  931  0.075
в разводе               1110   85  0.071
вдовец / вдова           896   63  0.066

**Вывод**

Зависимость есть, клиенты которые находятся в разводе или которых можно отнести к категории вдовец / вдова - чаще возвращают кредит в срок, чем клиенты в браке. Клиенты, которые не женаты / не замужем или находятся в гражданском браке, реже выплачивают кредит в срок, чем остальные категории. Это может быть связано с тем, что люди, находящиеся в браке сейчас или ранее, чувствуют большую ответственность и поэтому более дисциплинированны.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [27]:
data_pivot = data.pivot_table(index = ['income_group'], columns = 'debt', values = 'gender', aggfunc = 'count')

data_pivot['ratio'] = round(data_pivot[1] / (data_pivot[0] + data_pivot[1]), 3)
data_pivot.sort_values('ratio', ascending = False)

debt             0    1  ratio
income_group                  
2             4996  483  0.088
3             4799  448  0.085
1             4937  427  0.080
4             4980  383  0.071

**Вывод**

Зависимость есть, так клиент, чей ежемесячный доход больше, чем у 75% других клиентов из выборки, с наименьшей вероятностью просрочит платежи. При этом, затем идут клиенты из группы с наименьшим уровнем доходов, вероятно, они более организованны в финансах, чем клиенты, которые получают немного больше.

- Как разные цели кредита влияют на его возврат в срок?

In [28]:
data_pivot = data.pivot_table(index = ['purpose_group'], columns = 'debt', values = 'gender', aggfunc = 'count')

data_pivot['ratio'] = round(data_pivot[1] / (data_pivot[0] + data_pivot[1]), 3)
data_pivot.sort_values('ratio', ascending = False)

debt               0    1  ratio
purpose_group                   
автомобиль      3903  403  0.094
образование     3643  370  0.092
свадьба         2138  186  0.080
недвижимость   10028  782  0.072

**Вывод**

Среди четырех категорий, которые мы выделили: "свадьба", "недвижимость", "автомобиль", "образование", самая низкая вероятность задолженности у клиентов, которые берут кредит на недвижимость, большая вероятность задолженности у клиентов, которые берут кредиты на свадьбу, затем на образование и хуже всего платят клиенты по автокредитам. Это может быть связано с тем, что клиенты в большей степени заинтересованы в приобретении (покупке, строительстве, ремонте) собственного жилья.

## Шаг 4. Общий вывод

Таким образом, в результате анализа мы получили, что:
1. Наличие детей влияет на возврат кредита в срок, а именно: наличие детей у клиента увеличивает вероятность просроченной задолженности.
2. Семейное положение влияет на возврат кредита в срок, а именно: клиенты которые находятся в разводе или которых можно отнести к категории вдовец / вдова - чаще возвращают кредит в срок, чем клиенты в браке. Клиенты, которые не женаты / не замужем или находятся в гражданском браке, реже выплачивают кредит в срок, чем остальные категории.
3. Уровень дохода влияет на возврат кредита в срок:  клиенты, которые имеют относительно других самые высокие ежемесячные доходы с наименьшей вероятностью просрочат платежи, а также клиенты из группы с наименьшим уровнем доходов.
4. Цели кредита влияют на его возврат в срок, так: самая низкая вероятность задолженности у клиентов, которые берут кредит на недвижимость, большая вероятность задолженности у клиентов, которые берут кредиты на свадьбу, затем на образование и хуже всего платят клиенты по автокредитам.